internal/tiger/cmd/db.go

```diff
@@ -932,3 +960,363 @@ func isConnectionRejected(err error) bool {
 	// should be treated as "unreachable" (exit code 2)
 	return false
 }
+
+// isAuthenticationError checks if the error is a PostgreSQL authentication failure
+func isAuthenticationError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// Check for PostgreSQL error code 28P01 (invalid_password) or 28000 (invalid_authorization_specification)
+	var pgErr *pgconn.PgError
+	if errors.As(err, &pgErr) {
+		return pgErr.Code == "28P01" || pgErr.Code == "28000"
+	}
+	return false
+}
+
+// testConnectionWithPassword tests database connectivity with a specific password
+// Returns nil on success, error on failure
+func testConnectionWithPassword(ctx context.Context, details *common.ConnectionDetails, password string, timeout time.Duration) error {
+	// Build connection string with password (URL-encode the password to handle special chars)
+	connStr := fmt.Sprintf("postgresql://%s:%s@%s:%d/%s?sslmode=require",
+		details.Role, url.QueryEscape(password), details.Host, details.Port, details.Database)
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	conn, err := pgx.Connect(ctx, connStr)
+	if err != nil {
+		return err
+	}
+	defer conn.Close(ctx)
+	return nil
+}
+
+// passwordRecoveryOption represents the user's choice in the password recovery menu
+type passwordRecoveryOption int
+
+const (
+	optionEnterPassword passwordRecoveryOption = iota
+	optionResetPassword
+	optionExit
+)
+
+// passwordRecoveryModel is the Bubble Tea model for password recovery selection
+type passwordRecoveryModel struct {
```

Comment:
Might be nice to move this bubbletea model (and some of the related types/functions, like `passwordRecoveryOption` and `selectPasswordRecoveryOption`) out of `db.go` and into its own file. It feels like `db.go` is getting pretty big and unwieldy at this point.